src/contexts/LoginContext.js

```diff
@@ -1,3 +1,61 @@
-import { createContext } from 'react';
+import React, { createContext, useEffect, useState, useContext } from 'react';
+import axios from 'axios';
 
-export const LoginContext = createContext(null);
+const { REACT_APP_BACKEND_URL: BACKEND_URL } = process.env
+const LOCAL_STORAGE_USER_ID_KEY = "userId"
+
+const LoginContext = createContext(null);
+
+const LoginConsumer = ({ children }) => {
+  const loginContextData = useContext(LoginContext)
+  if (!loginContextData) throw new Error('useLoginContext must be used within an LoginProvider')
+
+  return <LoginContext.Consumer>{children}</LoginContext.Consumer>
+}
+
+const LoginProvider = ({ children }) => {
+  const [userId, setUserId] = useState(localStorage.getItem(LOCAL_STORAGE_USER_ID_KEY))
+
+  const verifyLoginAndSetLocalStorage = async () => {
+    const resp = await axios.get(`${BACKEND_URL}/auth/whoami`)
+    const { userId } = resp.data
+    setUserId(userId)
+    localStorage.setItem(LOCAL_STORAGE_USER_ID_KEY, userId)
+  }
+
+  const logout = async () => {
+    await axios.get(`${BACKEND_URL}/auth/logout`)
+    localStorage.removeItem(LOCAL_STORAGE_USER_ID_KEY)
+    setUserId(null)
+  }
```

Comment:
I believe this feature didn't exist iirc. Let's add it as a separate PR if so.

Created issue https://github.com/isomerpages/isomercms-frontend/issues/435

Reply:
Also we should put these toasts in the components that call the `logout()` function instead of inside `LoginContext.js`

Reply:
Hmm, I was thinking that we should put the toasts inside the `logout` function itself because unlike our other use cases, the logout functionality cuts across many layouts. But I guess in practice it would only exist in the `Sidebar` and the header in the `Sites` layout(?), so I'm ok with handling it there too

Reply:
As discussed, we can put the toasts in the catch statement of the `setRedirectToLogout` function from the `useRedirectHook` as the `logout` function should not handle view-layer logic for cleaner abstraction